### PR TITLE
Cleanup access rights middleware

### PR DESCRIPTION
**Pull request by furkansenharputlu**
`AccessRightCheck.ProcessRequest` looks complex. This PR cleans it up.

### CODE REVIEW

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5717/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6731291876)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5717/merge
Commit: 83c509925c4c8695cc34325756837c0dedc0724e 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6731466294)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5717/merge
Commit: f2a1a7a891bd26bb1eb6207e70c4069259a2258c 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6731564605)

File: gateway/mw_access_rights.go [hunk 1]
```diff
@@ -25,38 +25,32 @@ func (a *AccessRightsCheck) ProcessRequest(w http.ResponseWriter, r *http.Reques
...
+		return errors.New("Access to this API has been disallowed"), http.StatusForbidden
+	}
+
+	if a.Spec.VersionData.NotVersioned {
+		return nil, http.StatusOK
+	}
+
+	targetVersion := a.Spec.getVersionFromRequest(r)
+	if targetVersion == "" {
+		targetVersion = a.Spec.VersionData.DefaultVersion
```
Comment (titpetric) [inline gateway/mw_access_rights.go:45]: Should we move this to the end of the function ; if it's set it should be the fallback?

Comment (furkansenharputlu) [reply gateway/mw_access_rights.go:45]: but it is in the search below in the `for` loop.

**Review by titpetric — APPROVED** [review]

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5717/merge
Commit: 83c509925c4c8695cc34325756837c0dedc0724e 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6731775204)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5717/merge
Commit: 91ac64296185dcb541e42982facb283958295c20 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6735179253)

**Pull request merged**